src/dbnode/persist/fs/retriever.go

```diff
@@ -802,6 +801,9 @@ type retrieveRequest struct {
 	finalizes uint32
 	shard     uint32
 
+	finalized          uatomic.Bool
```

Comment:
make `finalizes` `uatomic.Uint32`-wrapped atomic, too, to prevent non-atomic use - esp. since you already fixed the misuse.